### PR TITLE
Support Cycle Unified

### PR DESCRIPTION
**Pull request by goodmind**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -14,23 +15,25 @@ export type Lens<T, R> = {
 }
 export type Scope<T, R> = string | number | Lens<T, R>;
 
-export function pick(selector: Selector | string) {
+export function pick<T>(selector: Selector | string) {
```
Comment (staltz) [inline src/index.ts]: Hi @goodmind, thanks for the PR.
Could you revert the changes in this part? I know it makes the typings generic for any stream library, but I don't want to make these typings less useful for xstream users. I think the approach should be like in Cycle DOM or Cycle HTTP, where RxJS users import `rxjs-typings.d.ts`. If that's not possible, then RxJS and most.js users should just use `any` casting.

File: src/index.ts [hunk 2]
```diff
@@ -14,23 +15,25 @@ export type Lens<T, R> = {
...
+      return adapt((xs.fromObservable(sinksArray$) as Stream<Array<any>>)
+        .map(sinksArray => sinksArray.map(selector)));
     };
   }
 }
 
-export function mix(aggregator: Aggregator) {
-  return function mixOperator(streamArray$: Stream<Array<Stream<any>>>): Stream<any> {
-    return streamArray$
+export function mix<T>(aggregator: Aggregator) {
```
Comment (staltz) [inline src/index.ts]: Same here.

File: src/index.ts [hunk 3]
```diff
@@ -115,7 +118,7 @@ export class StateSource<T> {
   public select<R>(scope: Scope<T, R>): StateSource<R> {
     const get = makeGetter(scope);
     return new StateSource<R>(
-      this.state$.map(get).filter(s => typeof s !== 'undefined'),
+      xs.fromObservable<T>(this.state$).map(get).filter(s => typeof s !== 'undefined'),
```
Comment (staltz) [inline src/index.ts]: We can avoid this one since RxJS/xstream/most.js all have `map` and `filter`. It should save us something in performance.

File: src/index.ts [hunk 4]
```diff
@@ -134,7 +137,8 @@ export default function onionify<So, Si>(
       .drop(1);
     sources[name] = new StateSource<any>(state$, name) as any;
     const sinks = main(sources as So);
-    reducerMimic$.imitate(sinks[name]);
+    const imitation$ = sinks[name] && xs.fromObservable(sinks[name]);
```
Comment (staltz) [inline src/index.ts]: I don't understand why this line is needed.

Comment (goodmind) [inline src/index.ts]: Because without it tests failing, there is two tests with undefined sink and when we do xs.fromObservable(undefined) it fails

File: src/index.ts (hunk 1 above)
Comment (goodmind) [inline src/index.ts]: Maybe we can use here generic defaults https://github.com/Microsoft/TypeScript/pull/13487

Comment (goodmind) [inline src/index.ts]: It will be even better if TypeScript supports Higher Kinded Types

File: src/index.ts (hunk 4 above)
Comment (staltz) [inline src/index.ts]: Okay. Can we use a traditional `if ( ) {` block here?

File: src/index.ts (hunk 1 above)
Comment (staltz) [inline src/index.ts]: We can when TypeScript 2.3 is out. https://github.com/Microsoft/TypeScript/milestone/38
We can either revert these changes, and later update to 2.3, or wait until 2.3 is out.

**Comment by goodmind** [comment]
@staltz any comments?

**Comment by staltz** [comment]
Looks good, thanks @goodmind

**Comment by staltz** [comment]
New version v3.2.0

File: src/index.ts (hunk 3 above)
Comment (goodmind) [inline src/index.ts]: According to https://github.com/staltz/cycle-onionify/issues/26 
This stream passed to constructor where `compose` called on it
